master/internal/experiment.go

```diff
@@ -484,25 +562,27 @@ func (e *experiment) isBestValidation(msg searcher.CompletedMessage) bool {
 	return isBest
 }
 
-func (e *experiment) updateState(ctx *actor.Context, state model.State) {
-	if wasPatched, err := e.Transition(state); err != nil {
+func (e *experiment) updateState(ctx *actor.Context, state model.State) (err error) {
+	var wasPatched bool
+	if wasPatched, err = e.Transition(state); err != nil {
 		ctx.Log().Error(err)
-		return
+		return err
```

Comment:
suggestion: e.Transition basically says if this is an illegal transition. One option, instead of returning an error, is to change this to a bool `ok` which states whether this transition is valid. That way we don't need to ignore a bunch of errors throughout experiment.go

Reply:
so return a bool ok instead of err and ignore those in all the same places?